main.js

```diff
@@ -135,9 +135,43 @@ options = {
 	port: config.minecraftserver.port,
 	version: config.minecraftserver.version
 }
-if (config.antiAntiAFK) setInterval(function () {
-	if(proxyClient == null && webserver.isInQueue && finishedQueue) client.write("chat", { message: "!que", position: 1 })
-}, 50000)
+
+const walkActions = [ 'forward', 'back', 'left', 'right'];
+function startAntiAntiAFK(){
+	if(!config.antiAntiAFK) return;
+	if(proxyClient == null && webserver.isInQueue && finishedQueue){
+		conn.bot.clearControlStates();
+
+		setTimeout(()=>{
+			if(conn.bot._client.state != "play"){
+				startAntiAntiAFK(); return;
+			}
+
+			var rotation, jump, walk;
+			while(!rotation && !jump && !walk){
+				rotation = (Math.random() < 0.75);
+				jump = (Math.random() < 0.75);
+				walk = (Math.random() < 0.75);
+			}
+			if(rotation){
+				var yaw = Math.random()*Math.PI - (0.5*Math.PI);
+				var pitch = Math.random()*Math.PI - (0.5*Math.PI);
+				conn.bot.look(yaw,pitch,false);
+			}
+			if(walk){
+				lastaction = walkActions[Math.floor(Math.random() * walkActions.length)];
+				conn.bot.setControlState(lastaction,true);
+			}
+			if(jump){
```

Comment:
jumping without eating will cause the player to die of hunger. As eating up
all the food is not a good idea either, you should remove jumping.

Reply:
Good point about jumping, it can make a little less effective(e.g. when bot get stuck in 1x1 hole), but it's probably better than diyng. As far as I know, var works only in function context, but I changed it to let as you requested

Reply:
>var works only in function context

you are right